File: shelfguard_app/lib/features/auth/domain/entities/shop.dart
import 'package:equatable/equatable.dart';

class Shop extends Equatable {
  final String id;
  final String name;
  final String? description;
  final String? address;
  final String? phone;
  final String? email;
  final String ownerId;
  final List<int> defaultNotificationDays;
  final String? notificationTime; // Format: "HH:mm"
  final bool notificationsEnabled;
  final DateTime createdAt;
  final DateTime updatedAt;

  const Shop({
    required this.id,
    required this.name,
    this.description,
    this.address,
    this.phone,
    this.email,
    required this.ownerId,
    required this.defaultNotificationDays,
    this.notificationTime,
    required this.notificationsEnabled,
    required this.createdAt,
    required this.updatedAt,
  });

  @override
  List<Object?> get props => [
        id,
        name,
        description,
        address,
        phone,
        email,
        ownerId,
        defaultNotificationDays,
        notificationTime,
        notificationsEnabled,
        createdAt,
        updatedAt,
      ];
}

File: shelfguard_backend/src/models/Shop.js
const { DataTypes } = require('sequelize');
const { sequelize } = require('../config/database');

const Shop = sequelize.define('Shop', {
  id: {
    type: DataTypes.UUID,
    defaultValue: DataTypes.UUIDV4,
    primaryKey: true,
  },
  name: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  description: {
    type: DataTypes.TEXT,
  },
  address: {
    type: DataTypes.TEXT,
  },
  phone: {
    type: DataTypes.STRING,
  },
  email: {
    type: DataTypes.STRING,
    validate: {
      isEmail: true,
    },
  },
  ownerId: {
    type: DataTypes.UUID,
    allowNull: false,
    field: 'owner_id',
  },
  defaultNotificationDays: {
    type: DataTypes.ARRAY(DataTypes.INTEGER),
    allowNull: false,
    defaultValue: [7, 3, 0],
    field: 'default_notification_days',
  },
  notificationTime: {
    type: DataTypes.STRING(5),
    field: 'notification_time',
  },
  notificationsEnabled: {
    type: DataTypes.BOOLEAN,
    allowNull: false,
    defaultValue: true,
    field: 'notifications_enabled',
  },
}, {
  tableName: 'shops',
  underscored: true,
  timestamps: true,
  createdAt: 'created_at',
  updatedAt: 'updated_at',
});

module.exports = Shop;
